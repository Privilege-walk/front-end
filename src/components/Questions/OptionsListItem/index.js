import React, { useState } from "react";
import {Input} from 'reactstrap';
import Button from '@mui/material/Button';

import Box from '@mui/material/Box';
import FormControl from '@mui/material/FormControl';


export default function OptionsListItem({ id, type, description, value, addOption, editOption, deleteOption, optionDisabled }) {
    const [optionDesc, setOptionDesc] = useState(description);
    const [optionPoints, setOptionPoints] = useState(value);
    const [isDisabled, setIsDisabled] = useState(optionDisabled);

    function validateOption() {
        let optiondesc = optionDesc === "" || optionDesc.trim() === "";
        let optionvalue = optionPoints === "" || optionPoints.trim() === "";
        return optiondesc || optionvalue;
    }

    function handleSubmit() {
        if (type === 'new') {
            addOption({
                id: id,
                description: optionDesc,
                value: optionPoints
            });
            setOptionDesc('');
            setOptionPoints('');
        } else {
            editOption({
                id: id,
                description: optionDesc,
                value: optionPoints
            });
            setIsDisabled(true);
        }
    }

    return (
        <div style={{p: '2px 4px', display: 'flex', flexGrow: 1, alignItems: 'center'}}>
            <FormControl sx={{ m: 1, width: '350px' }}>
                <Input
                    id={"new-option-desc-"+id}
                    data-testid={"new-option-desc-"+id}
                    placeholder="Option Description"
                    value={optionDesc}
                    onChange={(e) => setOptionDesc(e.target.value)}
                    disabled={isDisabled}
                />
            </FormControl>
            <FormControl sx={{ m: 1, width: '125px' }}>
                <Input
                    id={"new-option-points-"+id}
                    data-testid={"new-option-points-"+id}
                    placeholder="Option Points"
                    value={optionPoints}
                    onChange={(e) => setOptionPoints(e.target.value)}
                    disabled={isDisabled}
                />
            </FormControl>
            {
                (type === 'new' || (type === 'edit' && !isDisabled)) && 
                (<Box sx={{p: '0px 0px 0px 8px'}}>
                    <Button 
                        id={"add-"+id}
                        data-testid={"add-"+id}
                        variant="contained"
                        type="submit"
                        className="ml-2"
                        onClick={handleSubmit}
                        disabled={validateOption()}
                    >
                        {type === 'new' ? 'Add' : 'Submit'}
                    </Button>  
                </Box>)
            }
            {
                type === 'edit' && isDisabled && 
                (<Box sx={{p: '0px 0px 0px 8px'}}>
                    <Button 
                        id={"edit-"+id}
                        data-testid={"edit-"+id}
                        variant="contained"
                        type="submit"
                        className="ml-2"
                        onClick={() => setIsDisabled(false)}
                    >
                        Edit
                    </Button>  
                </Box>)
            }
            {
                type === 'edit' && isDisabled && 
                (<Box sx={{p: '0px 0px 0px 8px'}}>
                    <Button 
                        id={"delete-"+id}
                        data-testid={"delete-"+id}
                        variant="contained"
                        type="submit"
                        className="ml-2"
                        onClick={() => deleteOption(id)}
                    >
                        Delete
                    </Button>  
                </Box>)
            }
        </div>
    );
}