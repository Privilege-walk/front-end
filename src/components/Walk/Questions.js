import React from "react";
import Typography from '@mui/material/Typography';
import Grid from '@mui/material/Grid';
import Container from '@mui/material/Container';
import Button from '@mui/material/Button';
import Card from '@mui/material/Card';

import CardContent from '@mui/material/CardContent';


export default class Questions extends React.Component{
    
    constructor(props){
        super(props);
    }
    
    handleSelectChoice(choiceId, selected){
        let answer = selected? "" : choiceId;
        const props = this.props;
        let answers = { ...props.answers };
        let questionId = props.questions[props.questionIndex].id;
        answers[questionId] = answer;
        props.setAnswers(answers);
    }

    renderHeader(){
        return (
            <Grid container lg={1} item direction='column'>
                {/* Title */}
                <Grid item>
                    <Typography id="eventName"  variant="h5" component="div">
                        Event: {this.props.eventName}
                    </Typography>
                </Grid>

            {/* Graph */}
            {/* <Grid sx={{border: '2px solid'}} item>
                <Typography  variant="h5" component="div">
                </Typography>
            </Grid> */}
        </Grid>
        );
    }

    renderBody(){
        let question = ""; 
        let choices = [];
        let questionDetails = this.props.questions[this.props.questionIndex];
        if(questionDetails){
            question = questionDetails.description;
            choices = questionDetails.choices;
        }
        return (
            <Grid container lg={10} sx={{ minHeight: '225px'}} item direction='column'>
                {/* Question */}
                <Grid item>
                    <Typography id={"question"}  variant="h6" component="div">
                        {questionDetails? 
                            questionDetails.description : 
                            ""
                        }
                    </Typography>
                </Grid>

                {/* Choices */}
                <Grid 
                    sx={{mb: 1}} 
                    id="choices"
                    container item 
                    alignItems="flex-start" 
                    direction="column" 
                >
                    {choices.map((choice, _) => (
                        choice.id == this.props.answers[questionDetails.id]?
                            (
                                <Button
                                    variant="outlined" 
                                    onClick={() => this.handleSelectChoice(choice.id, true)}
                                    sx={{mx:1, mt:1}} 
                                    className="choice"
                                    style={{ border: '3px solid' }}
                                    key={choice.id}
                                >
                                    {choice.description}
                                </Button>
                            ):
                            (
                                <Button 
                                    variant="outlined" 
                                    className="choice"
                                    onClick={() => this.handleSelectChoice(choice.id, false)}
                                    sx={{mx:1, mt:1}} 
                                    key={choice.id}
                                >
                                    {choice.description}
                                </Button>
                            )
                        )
                    )}
                </Grid>
            </Grid>
        );
    }

    renderHostBody(){
        let question = ""; 
        let questionDetails = this.props.questions[this.props.questionIndex];
        if(questionDetails){
            question = questionDetails.description;
        }
        return (
            <Grid container lg={10} sx={{ minHeight: '225px'}} item direction='column'>
                {/* Question */}
                <Grid item>
                    <Typography id={"question"}  variant="h6" component="div">
                        {questionDetails? 
                            questionDetails.description : 
                            ""
                        }
                    </Typography>
                </Grid>
                <Grid container item>
                    <Card variant="outlined">
                    <CardContent>
                        <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
                            Total Participants
                        </Typography>
                        <Typography variant="h5" component="div">
                            {this.props.activeUsers}
                        </Typography>       
                    </CardContent>  
                    </Card>

                    <Card variant="outlined">
                    <CardContent>
                        <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
                            Total Answered
                        </Typography>
                        <Typography variant="h5" component="div">
                            {this.props.answeredUsers}
                        </Typography>       
                    </CardContent> 
                    </Card> 
                </Grid>
            </Grid>);
    }

    renderFooter(){
        const question = this.props.questions[this.props.questionIndex];
        const questionIndex = question? question.id: "";
        return (
            <Grid container lg={1} sx={{minHeight: '50px'}} item direction='column'>
                {/* Number of people who have answered so far */}
                {/* <Grid item>
                    <Typography  variant="h6" component="div"> 
                    </Typography>
                </Grid> */}

                <Grid container direction="column" item >

                    <Typography id="numQuestions"  variant="p" component="div">
                        {this.props.questionIndex + 1}/{this.props.questions.length} Questions
                    </Typography>
                    
                </Grid>

                {this.props.userType == "HOST"?
                    (
                        <Button 
                            variant={this.props.activeUsers == this.props.answeredUsers? "contained" : "outlined"}
                            type="submit"
                            sx={{mt:2}}
                            onClick={()=> this.props.nextQuestion()}                         
                        >
                            Next Question
                        </Button>
                    ) :
                    (
                        <Button 
                            variant="outlined"
                            type="submit"
                            sx={{mt:2}}
                            onClick={()=> {}}     
                            disabled={!this.props.answers[questionIndex]}        
                        >
                            
                            Waiting for host ...
                        </Button>
                    )
                        }  
            </Grid>
        );
    }

    render(){
        switch(this.props.userType){
            case 'HOST':
                return (
                    <Container 
                        sx={{px:6, mt:3}}
                    >
                        <Grid 
                            sx={{minHeight: '40px'}} 
                            container 
                            justifyContent='center' 
                            alignItems='center' 
                            direction='column' 
                            spacing={2}
                        ></Grid>
                        {this.renderHeader()}
                        {this.renderHostBody()}
                        {this.renderFooter()}
                    </Container>
                )
        }
        return (        
            <Container 
                sx={{px:6, mt:3}}
            >
                <Grid 
                    sx={{minHeight: '40px'}} 
                    container 
                    justifyContent='center' 
                    alignItems='center' 
                    direction='column' 
                    spacing={2}
                >
                    {/* Header */}
                    {this.renderHeader()}

                    {/* Body */}
                    {this.renderBody()}

                    {/* Footer */}
                    {this.renderFooter()}
                    
                </Grid>
            </Container>
        );
    }
}