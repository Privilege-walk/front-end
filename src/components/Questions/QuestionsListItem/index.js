import React, { useState } from "react";
 import Card from '@mui/material/Card';
 import CardContent from '@mui/material/CardContent';
 import Typography from '@mui/material/Typography';
 import OptionsListItem from "../OptionsListItem";


export default function QuestionsListItem({ description, choices }) {
    const [answerChoices, changeChoices] = useState([...choices].reverse())
    return (
        <Card className="">
            <CardContent>
            <Typography>{description}</Typography>
            </CardContent>
            {
                answerChoices.map( (item) => (
                    <OptionsListItem 
                        key={item.id}
                        type = 'view'
                        description = {item.description}
                        value = {item.value}
                        optionDisabled = {true}
                    />
                ))
            }
        </Card>
    );
}