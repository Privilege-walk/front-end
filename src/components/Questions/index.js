import React, { useState, useEffect } from "react";
import { connect } from 'react-redux';
import { useLocation } from "react-router-dom";
import {Row, Container} from 'reactstrap';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableRow from '@mui/material/TableRow';
import Paper from '@mui/material/Paper';
import Button from '@mui/material/Button';
import InputLabel from '@mui/material/InputLabel';
import TableHead from '@mui/material/TableHead';

import Box from '@mui/material/Box';
import FormControl from '@mui/material/FormControl';
import OutlinedInput from '@mui/material/OutlinedInput';
import QuestionsListItem from "./QuestionsListItem";
import OptionsListItem from "./OptionsListItem";
import { fetchQuestions, createQuestion } from '../../Store/actions';


function Questions({fetchQuestions, createQuestion}) {
    const props = useLocation();
    const [questionsList, setQuestionsList] = useState([]);
    const [optionsList, setOptionsList] = useState([]);
    const [newOptionId, setNewOptionId] = useState(0);
    const [newQuestion, setNewQuestion] = useState("");
    const [errMsg, setErrMsg] = useState("");

    useEffect(async () => fetchAllQuestions(), []);

    async function fetchAllQuestions() {
        const action = await fetchQuestions({ params: {event_id: props.state.id} });
        if (action.payload.error){
            setErrMsg("Unable to display the questions list!");
        }else{
            setQuestionsList(action.payload.questions);
        }
    }
    
    function createChoices() {
        let options = JSON.parse(JSON.stringify(optionsList));
        return options.map( item => {
            return { "description": item.description, "value": item.value };
        });
    }

    async function submitNewQuestion(event) {
        let requestBody = {
            "event_id": props.state.id,
            "title": newQuestion,
            "choices": createChoices()
        }
        const action = await createQuestion({ requestBody });
        if (action.payload.status == "created"){
            fetchAllQuestions();
            setNewQuestion(action.payload.newQuestion);
            setOptionsList(action.payload.optionsList);
            setNewOptionId(action.payload.newOptionId);
        }else{
            setErrMsg("Unable to create the question!");
        }
    }

    function handleOptionAdd(option) {
        let options = JSON.parse(JSON.stringify(optionsList));
        options.push(option);
        setOptionsList(options);
        setNewOptionId(newOptionId + 1);
    }

    function handleOptionEdit(option) {
        let options = JSON.parse(JSON.stringify(optionsList));
        options.forEach( (item, index) => {
            if(item.id === option.id) {
                options[index] = option;
            }
        });
        setOptionsList(options);
    }

    function handleOptionDelete(id) {
        let options = JSON.parse(JSON.stringify(optionsList));
        options = options.filter(x => x.id != id);
        setOptionsList(options);
    }

    return (
        <Container className="mx-2">
            <Row>
                <h1 className="event">{props.state.name}</h1>
                <Paper 
                    className="mb-3 align-items-center" 
                    variant="outlined" 
                    elevation={0} 
                    sx={{ p: '2px 4px', display: 'flex', flexGrow: 1, alignItems: 'center', flexDirection: 'column'}}
                >
                    <FormControl fullWidth sx={{ m: 1 }}>
                        <InputLabel htmlFor="new-question-name">Add New Question</InputLabel>
                        <OutlinedInput
                            id="new-question-name"
                            label="Add New Question"
                            variant="filled"
                            autoComplete="off"
                            value={newQuestion}
                            onChange={(e) => setNewQuestion(e.target.value)}
                        />
                    </FormControl>
                    <div style={{width: '100%'}}>
                        {
                            optionsList.map( (item) => (
                                <OptionsListItem 
                                    key={item.id}
                                    id = {item.id}
                                    type = 'edit'
                                    description = {item.description}
                                    value = {item.value}
                                    editOption = {handleOptionEdit}
                                    deleteOption = {handleOptionDelete}
                                    optionDisabled = {true}
                                />
                            ))
                        }
                        <OptionsListItem 
                            id = {newOptionId}
                            type = 'new'
                            description = ''
                            value = ''
                            addOption = {handleOptionAdd}
                            optionDisabled = {false}
                        />
                        <Box sx={{p: '20px 0px 8px 8px'}}>
                            <Button 
                                variant="contained"
                                type="submit"
                                className="ml-2"
                                onClick={submitNewQuestion}
                                disabled={(newQuestion === "" || newQuestion.trim() === "") || optionsList.length < 2}
                            >
                                Create Question
                            </Button>  
                        </Box>
                    </div>
                </Paper>
            </Row>

            <TableContainer component={Paper}>
                <Table sx={{ minWidth: 650 }} aria-label="simple table">
                    <TableHead>
                        <TableRow>
                            <TableCell>
                                <h4>Questions</h4>
                            </TableCell>
                        </TableRow>
                    </TableHead>
                    <TableBody>
                    {
                        questionsList.length === 0 ?
                        (<TableRow style={{color:'gray',marginLeft:10}}>
                            <TableCell><i>No questions to show.</i></TableCell>
                        </TableRow>)
                        :
                        (
                            questionsList.map( (item, _) => (
                                <TableRow key={item.id}>
                                    <TableCell>
                                        <QuestionsListItem 
                                            key={item.id}
                                            description={item.description}
                                            choices={item.choices}
                                        />
                                    </TableCell>
                                </TableRow>)
                        ))
                    }
                    </TableBody>
                </Table>
            </TableContainer>
        </Container>
    );
}

const mapStateToProps = state => {
    return {
        token: state.authToken
    };
};

const mapDispatchToProps = {
    fetchQuestions,
    createQuestion
};

var QuestionsContainer = connect(mapStateToProps, mapDispatchToProps)(Questions);
export default QuestionsContainer;