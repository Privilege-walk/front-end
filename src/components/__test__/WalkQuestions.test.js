import React from "react";
import { render, fireEvent, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

import Questions from '../Walk/Questions';

const mockedSetAnswers = jest.fn();
const mockedNextQuestion = jest.fn();
var questions = [
    {
        "id": 1,
        "description": "The Question content",
        "choices": [
            {
                "id": 1,
                "description": "Pizza",
                "value": 1
            },
            {
                "id": 2,
                "description": "Ice Cream",
                "value": 2
            },
            {
                "id": 3,
                "description": "Salt Water",
                "value": -1
            }
        ]
    }
];
var questionsStats = [
    {
        "question_id": 1,
        "answers": [
            {
                "answer_id": 1,
                "answer": "Pizza",
                "count": 0
            },
            {
                "answer_id": 2,
                "answer": "Ice Cream",
                "count": 0
            },
            {
                "answer_id": 3,
                "answer": "Salt Water",
                "count": 0
            }
        ]
    }
]
var props = {
    answers : {'1': 1},
    questions,
    questionIndex: 0,
    setAnswers: mockedSetAnswers,
    eventName: 'eventName',
    activeUsers: 1,
    answeredUsers: 1,
    nextQuestion: mockedNextQuestion,
    userType: 'HOST',
    currentPositions: {},
    questionsStats
}

afterEach(jest.clearAllMocks);

test('renders without crashing',  () => {
    let utils;
    act (() => {
        utils = render( <Questions {...props} />);
    }); 
    const { asFragment } = utils;
    expect(asFragment()).toMatchSnapshot();
})

test('when participant selects a choice', async () => {
    let mockProps = {...props, userType:'PARTICIPANT', answers : {}};
    let utils;
    act (() => {
        utils = render( <Questions {...mockProps} />);
    });
    const choiceBtn = screen.getByTestId("choice2");
    await act(async () => {   
        await fireEvent.click(choiceBtn);
    });
    const submitBtn = screen.getByText("Submit", {selector: "button"});
    await act(async () => {   
        await fireEvent.click(submitBtn);
    });
    expect(mockedSetAnswers).toHaveBeenCalledTimes(1);
    expect(mockedSetAnswers).toHaveBeenCalledWith({'1': 2});
})

test('when participant changes their choice multiple times', async () => {
    let mockProps = {...props, userType:'PARTICIPANT'};
    let utils;
    act (() => {
        utils = render( <Questions {...mockProps} />);
    });
    const choiceBtn1 = screen.getByTestId("choice1");
    await act(async () => {   
        await fireEvent.click(choiceBtn1);
    });
    const choiceBtn2 = screen.getByTestId("radio-label-2");
    await act(async () => {   
        await fireEvent.click(choiceBtn2);
    });
    const choiceBtn3 = screen.getByTestId("choice1");
    await act(async () => {   
        await fireEvent.click(choiceBtn3);
    });
    const submitBtn = screen.getByText("Submit", {selector: "button"});
    await act(async () => {   
        await fireEvent.click(submitBtn);
    });
    expect(mockedSetAnswers).toHaveBeenCalledTimes(1);
    expect(mockedSetAnswers).toHaveBeenCalledWith({'1': 1});
})

test('participant has empty questions list', () => {
    let mockProps = {...props, userType:'PARTICIPANT', questions:[], answers : {}};
    let utils;
    act (() => {
        utils = render( <Questions {...mockProps} />);
    });
    const text = screen.getByTestId("question");
    expect(text.value).toEqual(undefined);
})

test('when host clicks on next question', async () => {
    let mockProps = {...props};
    let utils;
    act (() => {
        utils = render( <Questions {...mockProps} />);
    });
    const nextQuesBtn = screen.getByText("Next Question", {selector: "button"});
    await act(async () => {   
        await fireEvent.click(nextQuesBtn);
    });
    expect(mockedNextQuestion).toHaveBeenCalledTimes(1);
})

test('host has empty questions list', () => {
    let mockProps = {...props, questions:[], answers : {}};
    let utils;
    act (() => {
        utils = render( <Questions {...mockProps} />);
    });
    const text = screen.getByTestId("question");
    expect(text.value).toEqual(undefined);
})