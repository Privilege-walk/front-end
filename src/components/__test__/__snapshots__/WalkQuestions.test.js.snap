// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1nz72na-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1n6nc7-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-lg-1 css-1j3clo9-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            id="eventName"
          >
            Event: eventName
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-lg-1 css-fit3tt-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            data-testid="question"
            id="question"
          >
            The Question content
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
              >
                Total Participants
              </p>
              <div
                class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
              >
                1
              </div>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
              >
                Total Answered
              </p>
              <div
                class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
              >
                1
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-1cylki8-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
        >
          <div
            class="MuiTypography-root MuiTypography-p css-37jdci-MuiTypography-root"
            id="numQuestions"
          >
            1/1 Questions
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1p8w7jk-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Next Question
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <span
          class="MuiBox-root css-msn52r"
        >
          Time - 00 : 00
        </span>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
      >
        <div
          class="MuiTypography-root MuiTypography-p css-37jdci-MuiTypography-root"
          id="question_graph"
        >
          The Question content
        </div>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-x9z34i-MuiContainer-root"
        >
          <div
            class="MuiGrid-root css-15dy41a-MuiGrid-root"
          >
            <div
              class="recharts-responsive-container"
              style="width: 100%; height: 100%;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
